battery_state_broadcaster/src/battery_state_broadcaster.cpp

```diff
@@ -0,0 +1,442 @@
+// Copyright (c) 2025, b-robotized Group
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "battery_state_broadcaster/battery_state_broadcaster.hpp"
+
+#include <limits>
+#include <memory>
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+#include "controller_interface/helpers.hpp"
+
+namespace battery_state_broadcaster
+{
+const auto kUninitializedValue = std::numeric_limits<double>::quiet_NaN();
+const size_t MAX_LENGTH = 64;
```

Comment:
```suggestion
const size_t MAX_LENGTH = 64;  // maximum length of strings to reserve
```
it's generally recommended to document magic numbers if the name isn't fully explanatory